coordinator/tributary/src/provided.rs

```diff
@@ -30,43 +30,27 @@ pub struct ProvidedTransactions<D: Db, T: Transaction> {
   pub(crate) transactions: HashMap<&'static str, VecDeque<T>>,
 }
 
-impl<D: Db, T: Transaction> ProvidedTransactions<D, T> {
-  fn transaction_key(&self, hash: &[u8]) -> Vec<u8> {
-    D::key(b"tributary_provided", b"transaction", [self.genesis.as_ref(), hash].concat())
-  }
-  fn current_provided_key(&self) -> Vec<u8> {
-    D::key(b"tributary_provided", b"current", self.genesis)
-  }
-  pub(crate) fn locally_provided_quantity_key(genesis: &[u8; 32], order: &str) -> Vec<u8> {
-    D::key(b"tributary_provided", b"local_quantity", [genesis, order.as_bytes()].concat())
-  }
-  pub(crate) fn on_chain_provided_quantity_key(genesis: &[u8; 32], order: &str) -> Vec<u8> {
-    D::key(b"tributary_provided", b"on_chain_quantity", [genesis, order.as_bytes()].concat())
-  }
-  pub(crate) fn block_provided_quantity_key(
-    genesis: &[u8; 32],
-    block: &[u8; 32],
-    order: &str,
-  ) -> Vec<u8> {
-    D::key(b"tributary_provided", b"block_quantity", [genesis, block, order.as_bytes()].concat())
-  }
-
-  pub(crate) fn on_chain_provided_key(genesis: &[u8; 32], order: &str, id: u32) -> Vec<u8> {
-    D::key(
-      b"tributary_provided",
-      b"on_chain_tx",
-      [genesis, order.as_bytes(), &id.to_le_bytes()].concat(),
-    )
+create_db!(
+  TributaryProvidedDb {
+    TransactionDb: (genesis: &[u8], hash: &[u8]) -> Vec<u8>,
+    CurrentDb: (genesis: &[u8]) -> Vec<u8>,
+    LocalQuantityDb: (genesis: &[u8], order: &[u8]) -> u32,
+    OnChainQuantityDb: (genesis: &[u8], order: &[u8]) -> u32,
+    BlockQuantityDb: (genesis: &[u8], block: &[u8], order: &[u8]) -> u32,
+    OnChainTxDb: (genesis: &[u8], order: &[u8], id: u32) -> [u8; 32]
```

Comment:
The first 4 dropped "Provided", the last one replaced it with "Tx". The first 4 are now ambiguous (quantity of *what*), and the last one loses specificity (as it's only for on-chain provided TXs, not for on-chain transactions in general). Mind if we keep "Provided"?